atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -229,266 +268,209 @@ public void dropTables(Set<TableReference> tableRefs) {
-        if (log.isTraceEnabled()) {
-            Stopwatch stopwatch = Stopwatch.createStarted();
-            delegate.put(tableRef, values, timestamp);
-            logCellsAndSize("put", tableRef.getQualifiedName(), values.keySet().size(), byteSize(values), stopwatch);
-        } else {
-            delegate.put(tableRef, values, timestamp);
-        }
+        maybeLog(() -> delegate.put(tableRef, values, timestamp),
+                slowLogPredicate,
+                (logger, stopwatch) -> logCellsAndSize(logger,
+                        "put", tableRef.getQualifiedName(), values.keySet().size(), byteSize(values), stopwatch));
     }
 
     @Override
     public void putMetadataForTable(TableReference tableRef, byte[] metadata) {
-        if (log.isTraceEnabled()) {
-            Stopwatch stopwatch = Stopwatch.createStarted();
-            delegate.putMetadataForTable(tableRef, metadata);
-            logTimeAndTable("putMetadataForTable", tableRef.getQualifiedName(), stopwatch);
-        } else {
-            delegate.putMetadataForTable(tableRef, metadata);
-        }
+        maybeLog(() -> delegate.putMetadataForTable(tableRef, metadata),
+                slowLogPredicate,
+                (logger, stopwatch) -> logTimeAndTable(logger,
+                        "putMetadataForTable", tableRef.getQualifiedName(), stopwatch));
     }
 
     @Override
     public void putMetadataForTables(Map<TableReference, byte[]> tableRefToMetadata) {
-        if (log.isTraceEnabled()) {
-            Stopwatch stopwatch = Stopwatch.createStarted();
-            delegate.putMetadataForTables(tableRefToMetadata);
-            logTimeAndTableCount("putMetadataForTables", tableRefToMetadata.keySet().size(), stopwatch);
-        } else {
-            delegate.putMetadataForTables(tableRefToMetadata);
-        }
+        maybeLog(() -> delegate.putMetadataForTables(tableRefToMetadata),
+                slowLogPredicate,
+                (logger, stopwatch) -> logTimeAndTableCount(logger,
+                        "putMetadataForTables", tableRefToMetadata.keySet().size(), stopwatch));
     }
 
     @Override
     public void putUnlessExists(TableReference tableRef, Map<Cell, byte[]> values) throws KeyAlreadyExistsException {
-        if (log.isTraceEnabled()) {
-            Stopwatch stopwatch = Stopwatch.createStarted();
-            delegate.putUnlessExists(tableRef, values);
-            logCellsAndSize("putUnlessExists", tableRef.getQualifiedName(), values.keySet().size(), byteSize(values), stopwatch);
-        } else {
-            delegate.putUnlessExists(tableRef, values);
-        }
+        maybeLog(() -> delegate.putUnlessExists(tableRef, values),
+                slowLogPredicate,
+                (logger, stopwatch) -> logCellsAndSize(logger,
+                        "putUnlessExists", tableRef.getQualifiedName(), values.keySet().size(), byteSize(values), stopwatch));
     }
 
+
+    //TODO shouldn't we log this one too?
```

Comment:
I don't believe this function is used frequently.  The only place I can think of it being used is on start up.  Let's remove this TODO for the time being.